packaging/macos/tests/test_packaging_assets.sh

```
@@ -0,0 +1,138 @@
+  bundle="$(cd "${bundle}" && pwd -P)"
+  mkdir -p "${bundle}/lib"
+  cp "${PKG_DIR}/install.sh"                              "${bundle}/install.sh"
+  cp "${PKG_DIR}/lib/installer_lib.sh"                    "${bundle}/lib/installer_lib.sh"
+  # Fake bundle-local plist + binary so the resolution has something to
+  # find. Contents don't matter, only existence.
+  printf '<?xml version="1.0"?><plist/>' > "${bundle}/com.defenseclaw.gateway.plist"
+  printf '#!/bin/sh\nexit 0\n' > "${bundle}/defenseclaw-gateway"
+  chmod 0755 "${bundle}/defenseclaw-gateway" "${bundle}/install.sh"
+
+  # Bypass the root check so we can reach the resolution block.
+  sed -i.bak 's/\[\[ \$EUID -eq 0 \]\] || die/[[ 1 -eq 0 ]] || : /' "${bundle}/install.sh"
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)